sorting/src/dip107/QuickSort.java

```diff
@@ -0,0 +1,55 @@
+package dip107;
+
+public class QuickSort implements Test {
+
+    public static void sort(int[] arr, int low, int high, int order) {
+        if (low < high) {
+            int part;
+            if (order == 1) {
+                part = partitionAscending(arr, low, high);
+            } else {
+                part = partitionDescending(arr, low, high);
+            }
+            sort(arr, low, part, order);
+            sort(arr, part + 1, high, order);
+        }
+    }
+
+    private static int partitionAscending(int[] arr, int low, int high) {
+        int pivot = arr[ (high + low) / 2 ];
+        int i = low - 1;
+        int j = high + 1;
+        while (true) {
+            do { i++; } while(arr[i] < pivot);
+            do { j--; } while(arr[j] > pivot);
+
+            if ( i >= j) { return j; }
+            swap(arr, i, j);
+        }
+    }
+
+
+    private static int partitionDescending(int[] arr, int low, int high) {
+        int pivot = arr[ (high + low) / 2 ];
+        int i = low - 1;
+        int j = high + 1;
+        while (true) {
+            do { i++; } while(arr[i] > pivot);
```

Comment:
Two methods, `partitionAscending` and `partitionDescending `differ in only 2 written characters. I advise passing a comparator to a single method `partition`, which could replace `while(arr[i] > pivot)` with `while(compare(arr[i], pivot))` and `while(arr[j] < pivot)` with `while(!compare(arr[j], pivot))`